src/state/Maps/selectors.js

```diff
@@ -422,36 +495,208 @@ const getAllLayersStateByMapKey = createCachedSelector(
+	let {key, name, opacity, styleKey, renderAsType, options: layerStateOptions} = layerState;
+
+	layerKey = layerKey || key;
+
+	// TODO temporary for development. Next, could be data source type rewritten in layer state (e.g. vector -> tiled-vector?)
+	if (renderAsType) {
+		type = renderAsType;
+	}
+
+	let options = {};
+	if (dataSourceOptions && layerStateOptions) {
+		options = {...dataSourceOptions, ...layerStateOptions};
+	} else if (layerStateOptions) {
+		options = {...layerStateOptions}
+	} else if (dataSourceOptions) {
+		options = {...dataSourceOptions};
+	}
+
+	if (type === 'wmts') {
+		options.url = dataSourceOptions.url || dataSourceOptions.urls[0];
+	} else if (type === 'wms') {
+		const {url, params, configuration, ...rest} = dataSourceOptions;
+		const singleTile = configuration && configuration.hasOwnProperty('singleTile') ? configuration.singleTile : false;
+
+		options = {
+			params: {
+				...params,
+				layers: rest.layers,
+				styles: rest.styles,
+			},
+			singleTile,
+			url
+		}
+	} else if (type === "vector" || type === "tiled-vector" ) {
+		let features, tiles = null;
+
+		if (type === "vector") {
+			features = DataSelectors.getFeatures(dataSource.key, fidColumnName, attributeDataSourceKeyAttributeKeyPairs);
+		} else if (type === "tiled-vector") {
+			const view = getViewByMapKeyObserver(mapKey);
+			const viewport = getViewportByMapKeyObserver(mapKey);
+			const tileList = helpers.getTiles(viewport.width, viewport.height, view.center, view.boxRange);
+			const level = helpers.getZoomLevel(viewport.width, viewport.height, view.boxRange);
+			tiles = DataSelectors.getTiles(dataSource.key, fidColumnName, level, tileList, spatialRelationsFilter, attributeRelationsFilter, attributeDataSourceKeyAttributeKeyPairs, styleKey);
+		}
+
+		let selected = null;
+		let style = options?.style;
+
+		if (options?.selected) {
+			selected = SelectionsSelectors.prepareSelectionByLayerStateSelected(options.selected);
+		}
+
+		if (!style && styleKey) {
+			style = StylesSelectors.getDefinitionByKey(styleKey);
+		}
+
+		options = {
+			...options,
+			...(selected && {selected}),
+			...(style && {style}),
+			...(features && {features}),
+			...(tiles && {tiles}),
+			fidColumnName,
+			geometryColumnName
+		};
+	}
+
+	return {
+		key: layerKey + '_' + index,
+		layerKey,
+		opacity: opacity || 1,
+		name,
+		type,
+		options
+	};
+});
+
+const getMapBackgroundLayer = createRecomputeSelector((mapKey, layerState) => {
+	if (!layerState) {
+		layerState = getBackgroundLayerStateByMapKeyObserver(mapKey);
+	}
+
+	if (layerState) {
+		if (layerState.type) {
+			return layerState;
+		} else {
+			// TODO filterByActive & metadata modifiers?
+			const layerKey = 'pantherBackgroundLayer';
+			const spatialDataSources = DataSelectors.spatialDataSources.getFiltered(layerState);
+			if (spatialDataSources) {
+				return spatialDataSources.map((dataSource, index) => getLayerByDataSourceAndLayerState(index, dataSource, layerState, layerKey));
```

Comment:
Yes - others are unneeded here -> undefined

Reply:
But yes, it only allows wms or wmts as background layers. TODO aded https://github.com/gisat-panther/ptr-state/pull/56/commits/25761fd7123efb4fed9b118cd8268772363665f6

Reply:
Can you add check if dataSource is type of wms or wmts inside  ```return spatialDataSources.map((dataSource, index) =>``` Otherwise in case of vector layer as a background layer it will fail somewhere.

Reply:
3ec47bc2f3fcfabcef5a2e68737adcb547938736